classes/01class/exercises/c01-e04/hsy3418/c01-e04.txt

```diff
@@ -0,0 +1,49 @@
+1. Create a S3 bucket by following https://docs.aws.amazon.com/AmazonS3/latest/gsg/CreatingABucket.html
+Major steps: Find the create bucket service and create a bucket by using
+- Name: academy-exe-bucket
+- Region: us-east-2
+2. Upload a picture to the bucket
+
+3. Create an IAM Role(https://aws.amazon.com/premiumsupport/knowledge-center/s3-instance-access-bucket/)
+- Get into IAM service
+- Create a role, type of trusted entity: AWS service
+- Choose EC2 to be the service will use this role
+- Click Next: Permissions, Next: Tags, and Next: Review.
+- role name: a_role
+- Create the role.
+- Add an inline policy to the role: under Visual editor tab
+  a. select S3 as a service
+  b. Action: ["s3：ListBucket", "s3:GetObject"]
+  c. Speficy ARN for bucket arn:aws:s3:::academy-exe-bucket 
+     for object arn:aws:s3:::academy-exe-bucket/*
+  d. policyName: s3_read_only   
+
+- Attach the IAM role to the EC2 instance
+  a. Go to EC2 console
+  b. Select Actions->instance settings->attach/replace IAM role 
+  c. Selec a_role
+
+- Connect to the EC2 
+a. $ssh -i ~/downloads/ec2test.pem ec2-user@ec2-18-222-169-134.us-east-2.compute.amazonaws.com
+b. create a ~/.aws folder
+c. create a file in under ~/.aws 
+d. Put below text into the file
+[profile a_role]
```

Comment:
You don't need to do this.. the instance role should work without any changes inside the instance.
Please try again and update your answer

Reply:
Ah yes, after deleting this file, it still works same.